framework/summary_model/src/lib.rs

```diff
@@ -0,0 +1,166 @@
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+use sha3::Digest;
+use std::collections::HashMap;
+use std::io::{BufRead, Read, Write};
+use std::path::PathBuf;
+
+/// Summary of a run
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct RunSummary {
+    /// The unique run id
+    ///
+    /// Chosen by the runner. Unique for each run.
+    pub run_id: String,
+    /// The name of the scenario that was run
+    pub scenario_name: String,
+    /// The time the run started
+    ///
+    /// This is a Unix timestamp in seconds.
+    pub started_at: i64,
+    /// The duration that the run was configured with, in seconds
+    ///
+    /// If the run was configured for soak testing, then this will not be set.
+    ///
+    /// It is possible that the run finished sooner than `started_at + run_duration` if all
+    /// behaviours failed. As long as [RunSummary::peer_end_count] is greater than 0 then that
+    /// number of agents will have run for the full duration.
+    pub run_duration: Option<u64>,
+    /// The number of peers configured
+    ///
+    /// This is the number of peers that were either configured or required by the behaviour
+    /// configuration.
+    ///
+    /// Note: This is only meaningful for single-conductor tests with the standard Wind Tunnel runner
+    /// or with the TryCP runner. In general, each node only sees the roles it was assigned and not
+    /// the roles that were assigned across the network.
+    pub peer_count: usize,
+    /// The number of peers at the end of the test
+    ///
+    /// If some peers exit early, for example due to a fatal error during a behaviour run or an
+    /// unavailable conductor, then this will be less than [RunSummary::peer_count].
+    ///
+    /// Note: This is only meaningful for single-conductor tests with the standard Wind Tunnel runner
+    /// or with the TryCP runner. In general, each node only sees the roles it was assigned and not
+    /// the roles that were assigned across the network.
+    pub peer_end_count: usize,
+    /// The behaviour configuration
+    ///
+    /// This is the number of agents that were assigned to each behaviour.
+    ///
+    /// Note: This is only meaningful for single-conductor tests with the standard Wind Tunnel runner
+    /// or with the TryCP runner. In general, each node only sees the roles it was assigned and not
+    /// the roles that were assigned across the network.
+    pub behaviours: HashMap<String, usize>,
```

Comment:
I'm not that keen on `_peer_count` because it's not just a count. I've renamed this to match what it's called in the runner `assigned_behaviours`. Better? If we were going to rename it to anything else here, I'd want to rename it in the runner too